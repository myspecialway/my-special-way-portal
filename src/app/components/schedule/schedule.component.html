<table class="mat-table schedule-table" role="grid">
  <thead>
    <tr class="mat-header-row" role="row">
      <th class="mat-header-cell cdk-column-position mat-column-position" role="columnheader" colspan="2">שעה</th>
      <th class="mat-header-cell cdk-column-position mat-column-position" role="columnheader" *ngFor="let day of daysLabels">{{day}}</th>
    </tr>
  </thead>
  <tbody>
    <tr class="mat-row" role="row" *ngFor="let row of schedule; index as hourIndex">
      <td class="mat-cell cdk-column-position mat-column-position" role="gridcell">{{hourIndex + 1}}</td>
      <td class="mat-cell cdk-column-position mat-column-position" role="gridcell">{{hoursLabels[hourIndex]}}</td>
      <td 
        class="mat-cell cdk-column-position mat-column-position"
        *ngFor="let timeslot of row; index as dayIndex" 
        role="gridcell" 
        (click)="timeSlotClicked.emit({hourIndex: hourIndex, dayIndex: dayIndex})"
      >
        <div *ngIf="timeslot.lesson" class="ts-full">
          <span class="ts-lesson">{{timeslot.lesson.title}}</span>
          <span class="ts-location" *ngIf="timeslot.location">{{timeslot.location.name}}</span>
        </div>
        <div *ngIf="!timeslot.lesson" class="ts-empty">
          <i class="material-icons" >
            add
          </i>
        </div>
      </td>
    </tr>
  </tbody>
</table>
