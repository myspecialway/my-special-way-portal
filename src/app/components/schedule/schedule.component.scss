$hover: rgba(106, 0, 207, 0.2);

.schedule-table {
    width: 100%;
    direction: rtl;
    font-family: "Rubik", "Helvetica", "Arial", sans-serif;
    color: #222;
    th.mat-header-cell {
        background-color: #fafafa;
        font-size: 16px;
        font-weight: bold;
        color: #222;
        border: 0;
    }
    td.mat-cell, td.mat-footer-cell, th.mat-header-cell {
        text-align: right;
        padding: 20px 14px;
    }
    td.mat-cell, td.mat-footer-cell {
        cursor: pointer;
        border: 1px solid #e0e0e0;
        font-size: 18px;
        &:hover {
            background-color: $hover;
        }
    }

    td.mat-cell:first-child, td:nth-child(2) {
        text-align: center;
        direction: initial;
        font-weight: 400;
    }
    
    .ts-lesson, .ts-location {
        font-family: Rubik;
        color: #222;
        display: block;
    }
    .ts-lesson {
        font-size: 18px;
        font-weight: 400;
    }
    .ts-location {
        font-size: 14px;
    }
    .ts-empty {
        text-align: center;
        color: #6a00cf;
        opacity: .5;
        &:hover {
            opacity: 1;
        }
    }
}
