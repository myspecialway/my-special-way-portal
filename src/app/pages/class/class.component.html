<div class="main-content msw-class-container" data-test-id="classes-page">
  <div class="container-fluid">
    <div class="header-menu-container msw-table-tools-header">
      <div class="header-table-count">
        <div>
          <span class="header-table-count-label">כיתות</span>
          <span class="header-table-count-number">({{dataSource.data.length}})</span>
        </div>
      </div>
      <div class="example-header msw-table-tools-header">
        <button data-test-id="add-class-button" mat-raised-button color="primary" [routerLink]="['/class', '_new_']">הוסף כיתה חדשה</button>
      </div>
    </div>
    <div class="row msw-table-container">

      <div class="col-lg-12 col-md-12 col-sm-12">
        <mat-table #table [dataSource]="dataSource" class="msw-table"
                   matSort matSortActive="name" matSortDisableClear matSortDirection="asc">

          <!-- Number Column -->
          <ng-container matColumnDef="classname">
            <mat-header-cell *matHeaderCellDef>שם כיתה</mat-header-cell>
            <mat-cell *matCellDef="let row" (click)="navigate(row._id)" class="class-name">{{ row.name }}</mat-cell>
          </ng-container>

          <!-- Title Column -->
          <ng-container matColumnDef="level">
            <mat-header-cell *matHeaderCellDef>כיתה</mat-header-cell>
            <mat-cell *matCellDef="let row" (click)="navigate(row._id)">{{ scheduleService.grades[row.grade] }}</mat-cell>
          </ng-container>

          <!-- State Column-->
          <ng-container matColumnDef="deleteClass">
            <mat-header-cell *matHeaderCellDef></mat-header-cell>
            <mat-cell *matCellDef="let row">
              <button class="delete-row-button" [attr.data-test-id]="'delete-class-button-' + row.name" mat-icon-button color="primary" (click)="deleteClass(row._id, row.name, row.level, row.students.length)"><i class="material-icons"> delete </i></button>
            </mat-cell>
          </ng-container>

          <mat-header-row *matHeaderRowDef="displayedColumns"></mat-header-row>
          <mat-row *matRowDef="let row; columns: displayedColumns"></mat-row>
        </mat-table>
      </div>
    </div>
  </div>
</div>


<!-- Copyright 2018 Google Inc. All Rights Reserved.
    Use of this source code is governed by an MIT-style license that
    can be found in the LICENSE file at http://angular.io/license -->
