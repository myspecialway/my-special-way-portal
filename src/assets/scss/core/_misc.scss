body {
  background-color: #fafafa;
  color: $black-color;
  font-weight: 300;
}

legend {
  border-bottom: 0;
}

.serif-font {
  font-family: $font-family-serif;
}

// Prevent highlight on mobile
* {
  -webkit-tap-highlight-color: rgba(255, 255, 255, 0);
  -webkit-tap-highlight-color: transparent;
  &:focus {
    outline: 0;
  }
}

a {
  color: $link-color;
  &:hover,
  &:focus {
    color: darken($link-color, 5%);
    text-decoration: none;
  }

  &.text-info {
    &:hover,
    &:focus {
      color: darken($brand-info, 5%);
    }
  }

  & .material-icons {
    vertical-align: middle;
  }
}

.form-check,
label {
  font-size: 14px;
  line-height: 1.42857;
  color: $checkboxes-text-color;
  font-weight: 400;
}

/*           Animations              */
.animation-transition-general {
  @include transition-all($general-transition-time, $transition-linear);
}

.animation-transition-slow {
  @include transition-all($slow-transition-time, $transition-linear);
}

.animation-transition-fast {
  @include transition-all($fast-transition-time, $transition-ease);
}

.caret,
.sidebar a {
  @include transition-all($fast-transition-time, $transition-ease-in);
}
#map {
  position: relative;
  z-index: 2;
  width: 100%;
  height: calc(100vh - 70px);
  margin-top: 70px;
}

@font-face {
  font-family: 'pass';
  font-style: normal;
  font-weight: 400;
  src: url(data:application/font-woff;charset=utf-8;base64,d09GRgABAAAAAATsAA8AAAAAB2QAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAABGRlRNAAABWAAAABwAAAAcg9+z70dERUYAAAF0AAAAHAAAAB4AJwANT1MvMgAAAZAAAAA/AAAAYH7AkBhjbWFwAAAB0AAAAFkAAAFqZowMx2N2dCAAAAIsAAAABAAAAAQAIgKIZ2FzcAAAAjAAAAAIAAAACAAAABBnbHlmAAACOAAAALkAAAE0MwNYJ2hlYWQAAAL0AAAAMAAAADYPA2KgaGhlYQAAAyQAAAAeAAAAJAU+ATJobXR4AAADRAAAABwAAAAcCPoA6mxvY2EAAANgAAAAEAAAABAA5gFMbWF4cAAAA3AAAAAaAAAAIAAKAE9uYW1lAAADjAAAARYAAAIgB4hZ03Bvc3QAAASkAAAAPgAAAE5Ojr8ld2ViZgAABOQAAAAGAAAABuK7WtIAAAABAAAAANXulPUAAAAA1viLwQAAAADW+JM4eNpjYGRgYOABYjEgZmJgBEI2IGYB8xgAA+AANXjaY2BifMg4gYGVgYVBAwOeYEAFjMgcp8yiFAYHBl7VP8wx/94wpDDHMIoo2DP8B8kx2TLHACkFBkYA8/IL3QB42mNgYGBmgGAZBkYGEEgB8hjBfBYGDyDNx8DBwMTABmTxMigoKKmeV/3z/z9YJTKf8f/X/4/vP7pldosLag4SYATqhgkyMgEJJnQFECcMOGChndEAfOwRuAAAAAAiAogAAQAB//8AD3jaY2BiUGJgYDRiWsXAzMDOoLeRkUHfZhM7C8Nbo41srHdsNjEzAZkMG5lBwqwg4U3sbIx/bDYxgsSNBRUF1Y0FlZUYBd6dOcO06m+YElMa0DiGJIZUxjuM9xjkGRhU2djZlJXU1UDQ1MTcDASNjcTFQFBUBGjYEkkVMJCU4gcCKRTeHCk+fn4+KSllsJiUJEhMUgrMUQbZk8bgz/iA8SRR9qzAY087FjEYD2QPDDAzMFgyAwC39TCRAAAAeNpjYGRgYADid/fqneL5bb4yyLMwgMC1H90HIfRkCxDN+IBpFZDiYGAC8QBbSwuceNpjYGRgYI7594aBgcmOAQgYHzAwMqACdgBbWQN0AAABdgAiAAAAAAAAAAABFAAAAj4AYgI+AGYB9AAAAAAAKgAqACoAKgBeAJIAmnjaY2BkYGBgZ1BgYGIAAUYGBNADEQAFQQBaAAB42o2PwUrDQBCGvzVV9GAQDx485exBY1CU3PQgVgIFI9prlVqDwcZNC/oSPoKP4HNUfQLfxYN/NytCe5GwO9/88+/MBAh5I8C0VoAtnYYNa8oaXpAn9RxIP/XcIqLreZENnjwvyfPieVVdXj2H7DHxPJH/2/M7sVn3/MGyOfb8SWjOGv4K2DRdctpkmtqhos+D6ISh4kiUUXDj1Fr3Bc/Oc0vPqec6A8aUyu1cdTaPZvyXyqz6Fm5axC7bxHOv/r/dnbSRXCk7+mpVrOqVtFqdp3NKxaHUgeod9cm40rtrzfrt2OyQa8fppCO9tk7d1x0rpiQcuDuRkjjtkHt16ctbuf/radZY52/PnEcphXpZOcofiEZNcQAAeNpjYGIAg///GBgZsAF2BgZGJkZmBmaGdkYWRla29JzKggxD9tK8TAMDAxc2D0MLU2NjENfI1M0ZACUXCrsAAAABWtLiugAA)
    format('woff');
}

input[type='password'] {
  font-family: 'pass', 'Roboto', Helvetica, Arial, sans-serif;
  font-size: 18px;
  &::-webkit-input-placeholder {
    transform: scale(0.77);
    transform-origin: 0 50%;
  }
  &::-moz-placeholder {
    font-size: 14px;
    opacity: 1;
  }
  &:-ms-input-placeholder {
    font-size: 14px;
    font-family: 'Roboto', Helvetica, Arial, sans-serif;
  }
}
