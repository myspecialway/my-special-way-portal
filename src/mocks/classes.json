[
  {
    "_id": "5b0116f8c1d99ef1b8e73245",
    "index": 0,
    "grade": "A",
    "number": 1,
    "description": "Tiltan",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f870a67d3a47e9f235",
    "index": 1,
    "grade": "A",
    "number": 2,
    "description": "Tapuz",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f82074861f963642f3",
    "index": 2,
    "grade": "A",
    "number": 3,
    "description": "Tapuch",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f8b9e91d54d315372c",
    "index": 3,
    "grade": "A",
    "number": 4,
    "description": "Tut",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f8895af89de5563236",
    "index": 4,
    "grade": "B",
    "number": 1,
    "description": "Dafna",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f870788b1e9f99b814",
    "index": 5,
    "grade": "B",
    "number": 2,
    "description": "Dekel",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f896b61b3a5c135a1d",
    "index": 6,
    "grade": "B",
    "number": 3,
    "description": "Duvdevan",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f855ecf6fd19ae77fa",
    "index": 7,
    "grade": "B",
    "number": 4,
    "description": "Dolev",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f814169898640db32f",
    "index": 8,
    "grade": "C",
    "number": 1,
    "description": "Egoz",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f814169898640db32f",
    "index": 8,
    "grade": "C",
    "number": 2,
    "description": "Agas",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f814169898640db32f",
    "index": 8,
    "grade": "C",
    "number": 3,
    "description": "Erez",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f814169898640db32f",
    "index": 8,
    "grade": "C",
    "number": 4,
    "description": "Oren",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f814169898640db32f",
    "index": 8,
    "grade": "C",
    "number": 5,
    "description": "Alon",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f87554b3534cfde9b7",
    "index": 9,
    "grade": "D",
    "number": 1,
    "description": "Shaked",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f8cc498da37f361b2f",
    "index": 10,
    "grade": "D",
    "number": 2,
    "description": "Gefen",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f832a6de313076c3b1",
    "index": 11,
    "grade": "D",
    "number": 3,
    "description": "Shezif",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f832a6de313076c3b1",
    "index": 11,
    "grade": "D",
    "number": 4,
    "description": "Shesek",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f841bc308bff51f0b8",
    "index": 12,
    "grade": "E",
    "number": 1,
    "description": "Rimon",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f8897ec61406fc591f",
    "index": 13,
    "grade": "E",
    "number": 2,
    "description": "Teena",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f867a266b544055fc3",
    "index": 14,
    "grade": "E",
    "number": 3,
    "description": "Zait",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f846bd827beedec579",
    "index": 15,
    "grade": "F",
    "number": 1,
    "description": "Banana",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f8ab446a3271359472",
    "index": 16,
    "grade": "F",
    "number": 2,
    "description": "Balon",
    "num_of_students": 20
  },
  {
    "_id": "5b0116f8f0319d311adbd5b8",
    "index": 17,
    "grade": "F",
    "number": 3,
    "description": "Bait",
    "num_of_students": 20
  }
]
