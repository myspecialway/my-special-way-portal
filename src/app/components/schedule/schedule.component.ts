import { Component, Input, Output, EventEmitter } from '@angular/core';
import { TimeSlot } from '../../models/timeslot.model';

export interface TimeSlotIndexes {
  hourIndex: number;
  dayIndex: number;
}

@Component({
  selector: 'app-schedule',
  templateUrl: './schedule.component.html',
  styleUrls: ['./schedule.component.scss'],
})

export class ScheduleComponent {

  @Input()
  schedule: TimeSlot[][];
  @Input()
  daysLabels: string[];
  @Input()
  hoursLabels: string[];
  @Output()
  timeSlotClicked: EventEmitter<TimeSlotIndexes> = new EventEmitter();
}
