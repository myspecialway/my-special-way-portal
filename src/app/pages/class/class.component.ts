import { Component, OnInit, ElementRef, ViewChild } from '@angular/core';
import 'rxjs/add/observable/merge';
import 'rxjs/add/observable/fromEvent';
import 'rxjs/add/operator/map';
import 'rxjs/add/operator/debounceTime';
import 'rxjs/add/operator/distinctUntilChanged';
import { MatTableDataSource, MatSort, MatDialog } from '@angular/material';
import { ClassService } from './services/class.graphql.service';
import { Class } from '../../models/class.model';
import { ScheduleService } from '../../services/schedule/schedule.service';
import { MSWSnackbar } from '../../services/msw-snackbar/msw-snackbar.service';
import { DeleteClassDialogComponent } from './dialogs/delete/delete-class.dialog';
import { Router } from '@angular/router';

@Component({
  selector: 'app-grade',
  templateUrl: './class.component.html',
  styleUrls: ['./class.component.scss'],
})
export class ClassComponent implements OnInit {
  displayedColumns = ['classname', 'level', 'deleteClass'];
  dataSource = new MatTableDataSource<Class>();
  resultsLength = 0;

  @ViewChild(MatSort)
  sort: MatSort;
  @ViewChild('table')
  table: ElementRef;

  constructor(
    private classService: ClassService,
    public dialog: MatDialog,
    public scheduleService: ScheduleService,
    private mswSnackbar: MSWSnackbar,
    private router: Router,
  ) {}

  async ngOnInit() {
    await this.populateDatasource();
    this.dataSource.sort = this.sort;
  }

  private async populateDatasource() {
    try {
      this.classService.getAllClasses().subscribe((classes) => {
        this.dataSource.data = [...classes];
      });
    } catch (error) {
      // TODO: implement error handling on UI
      console.error('Error handling not implemented');
      throw error;
    }
  }

  deleteClass(_id: string, name: string, level: string, numberOfStudents: number) {
    if (numberOfStudents > 0) {
      this.mswSnackbar.displayTimedMessage('לא ניתן למחוק את הכיתה כיוון שיש תלמידים המשוייכים אליה');
    } else {
      const dialogRef = this.dialog.open(DeleteClassDialogComponent, {
        data: { _id, name, level },
      });
      dialogRef.afterClosed().subscribe(async (result) => {
        if (result === true) {
          await this.classService.delete(_id);
        }
      });
    }
  }

  navigate(row_id: string) {
    this.router.navigate(['/class', row_id]);
  }
}
