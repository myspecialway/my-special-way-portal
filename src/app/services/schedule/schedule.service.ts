import { Injectable } from '@angular/core';

@Injectable()
export class ScheduleService {
  levels = ['א', 'ב', 'ג', 'ד', 'ה', 'ו'];
  daysLabels = ['ראשון', 'שני', 'שלישי', 'רביעי', 'חמישי', 'שישי'];
  hoursLabels = [
    '07:30 - 08:00',
    '08:00 - 08:50',
    '08:50 - 09:35',
    '09:35 - 10:00',
    '10:00 - 10:30',
    '10:30 - 11:15',
    '11:15 - 12:00',
    '12:00 - 12:45',
    '12:45 - 13:15',
    '13:15 - 13:45',
    '13:45 - 14:30',
    '14:30 - 15:15',
    '15:15 - 16:00',
    '16:00 - 16:30',
    '16:30 - 16:45',
  ];
}
