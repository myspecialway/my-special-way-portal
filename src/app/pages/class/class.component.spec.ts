import { ClassComponent } from './class.component';
import { TestBed, ComponentFixture } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { MatHeaderRow, MatRowDef, MatHeaderRowDef, MatSort, MatDialog } from '@angular/material';
import { ClassService } from './services/class.graphql.service';
import { classTestData } from '../../../mocks/assets/classes.mock';
import {
  Overlay,
  ScrollStrategyOptions,
  ScrollDispatcher,
  OverlayKeyboardDispatcher,
  OverlayPositionBuilder,
  OverlayContainer,
  ViewportRuler,
} from '@angular/cdk/overlay';
import { Platform } from '@angular/cdk/platform';
import { Observable } from 'rxjs-compat';
import { ScheduleService } from '../../services/schedule/schedule.service';
import { MSWSnackbar } from '../../services/msw-snackbar/msw-snackbar.service';
import { Router } from '@angular/router';
import 'rxjs-compat/add/observable/of';

describe('class component', () => {
  let classServiceMock: Partial<ClassService>;
  let classDialogMock: Partial<MatDialog>;
  let scheduleServiceMock: Partial<ScheduleService>;
  let snackbarMock: Partial<MSWSnackbar>;
  let routeMock: Partial<Router>;
  let fixture: ComponentFixture<ClassComponent>;
  let component: ClassComponent;

  beforeEach(async () => {
    classServiceMock = {
      getAllClasses: jest.fn().mockReturnValueOnce(Observable.of(classTestData.classes)),
      delete: jest.fn(),
      create: jest.fn(),
      update: jest.fn(),
    };

    classDialogMock = {
      open: jest.fn().mockReturnValue({
        afterClosed: jest.fn().mockReturnValue(Observable.of(true)),
      }),
    };

    snackbarMock = {
      displayTimedMessage: jest.fn(),
    };

    scheduleServiceMock = {
      grades: {
        a: 'א',
        b: 'ב',
        c: 'ג',
        d: 'ד',
        e: 'ה',
        f: 'ו',
      },
    };

    routeMock = {
      navigate: jest.fn(),
    };

    TestBed.configureTestingModule({
      imports: [],
      declarations: [ClassComponent, MatHeaderRow, MatRowDef, MatHeaderRowDef, MatSort],
      providers: [
        { provide: MatDialog, useValue: classDialogMock },
        { provide: ClassService, useValue: classServiceMock },
        { provide: ScheduleService, useValue: scheduleServiceMock },
        { provide: MSWSnackbar, useValue: snackbarMock },
        { provide: Router, useValue: routeMock },
        Overlay,
        ScrollStrategyOptions,
        ScrollDispatcher,
        Platform,
        ViewportRuler,
        OverlayContainer,
        OverlayPositionBuilder,
        OverlayKeyboardDispatcher,
      ],
      schemas: [NO_ERRORS_SCHEMA],
    });

    fixture = TestBed.createComponent(ClassComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should render component as described in snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should open dialog when calling deleteClass function when no students assigned to the class', () => {
    (classServiceMock.delete as jest.Mock).mockImplementationOnce(() => {
      return Promise.resolve(1);
    });

    component.deleteClass('123', 'ddd', 'ddd', 0);
    const DialogMock = TestBed.get(MatDialog);
    expect(DialogMock.open).toHaveBeenCalled();
  });

  it('should not open a dialog when calling deleteClass function when students are assigned to the class', () => {
    (classServiceMock.delete as jest.Mock).mockImplementationOnce(() => {
      return Promise.resolve(0);
    });
    component.deleteClass('123', 'ddd', 'ddd', 1);
    const DialogMock = TestBed.get(MatDialog);
    expect(DialogMock.open).not.toHaveBeenCalled();
  });

  it('should open a snackbar error message when calling deleteClass function when students are assigned to the class', () => {
    (classServiceMock.delete as jest.Mock).mockImplementationOnce(() => {
      return Promise.resolve(0);
    });
    component.deleteClass('123', 'ddd', 'ddd', 1);
    const SnackbarMock = TestBed.get(MSWSnackbar);
    expect(SnackbarMock.displayTimedMessage).toHaveBeenCalled();
  });

  it('should load classes from service on page load ', () => {
    (classServiceMock.getAllClasses as jest.Mock).mockImplementationOnce(() => {
      return Promise.resolve(classTestData.classes);
    });
    const DialogMock = TestBed.get(ClassService);
    expect(DialogMock.getAllClasses).toHaveBeenCalled();
  });

  it('should load correct number of classes', async () => {
    (classServiceMock.getAllClasses as jest.Mock).mockImplementationOnce(() => {
      return Promise.resolve(classTestData.classes);
    });
    await fixture.whenRenderingDone();
    expect(component.dataSource.data.length).toEqual(23);
  });
  it('should load zero classes in case of promise reject', async () => {
    const getAllClassesMock = classServiceMock.getAllClasses as jest.Mock;
    getAllClassesMock.mockReset();
    getAllClassesMock.mockImplementationOnce(() => Observable.of([]));

    // await fixture.whenRenderingDone();
    component.ngOnInit();
    expect(component.dataSource.data.length).toEqual(0);
  });

  it('should open edit-class-dialog ', () => {
    component.navigate('class/some-class-id');
    expect(routeMock.navigate).toHaveBeenCalled();
  });
});
