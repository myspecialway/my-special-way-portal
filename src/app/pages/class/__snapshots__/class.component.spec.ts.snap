// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class component should render component as described in snapshot 1`] = `
<app-grade
  classService={[Function Object]}
  dataSource={[Function MatTableDataSource]}
  dialog={[Function Object]}
  displayedColumns={[Function Array]}
  mswSnackbar={[Function Object]}
  resultsLength="0"
  router={[Function Object]}
  scheduleService={[Function Object]}
  sort={[Function MatSort]}
  subCollector={[Function Subscription]}
  table={[Function ElementRef]}
  translate={[Function TranslateService]}
>
  <div
    class="main-content msw-class-container"
    data-test-id="classes-page"
  >
    <div
      class="container-fluid"
    >
      <div
        class="header-menu-container msw-table-tools-header"
      >
        <div
          class="header-table-count"
        >
          <div>
            <span
              class="header-table-count-label"
            >
               CLASS.CLASSES
            </span>
            <span
              class="header-table-count-number"
            >
              (23)
            </span>
          </div>
        </div>
        <div
          class="example-header msw-table-tools-header"
        >
          <button
            color="primary"
            data-test-id="add-class-button"
            mat-raised-button=""
          >
             CLASS.NEW_CLASS_BTN 
          </button>
        </div>
      </div>
      <div
        class="row msw-table-container"
      >
        <div
          class="col-lg-12 col-md-12 col-sm-12"
        >
          <mat-table
            class="msw-table"
            matsort=""
            matsortactive="name"
            matsortdirection="asc"
            matsortdisableclear=""
            ng-reflect-active="name"
            ng-reflect-direction="asc"
            ng-reflect-disable-clear=""
          >
            
            
            
            
            
            
            
            
            
            
            
          </mat-table>
        </div>
      </div>
    </div>
  </div>
</app-grade>
`;
